src/flagsmith/mod.rs

```diff
@@ -151,12 +153,8 @@ impl Flagsmith {
                     Err(TryRecvError::Empty) => {}
                 }
 
-                let environment = Some(
-                    get_environment_from_api(&client, environment_url.clone())
-                        .expect("updating environment document failed"),
-                );
-                let mut data = ds.lock().unwrap();
-                data.environment = environment;
+                update_environment(&client, &ds, &environment_url).unwrap();
```

Comment:
Interested to know why we use `client.clone()` [above](https://github.com/Flagsmith/flagsmith-rust-client/pull/20/files#diff-61d6a1ad7f2c7844496bd8cd330365681bbdc9a0ccea9dd5a844e6bc408d1195R118) but directly point to the `client` here. 

I suspect it's because we want to update the environment on the instance of the client itself, but just double checking.

Reply:
I wouldn't want to clone the client struct on every background update.

Reply:
I'm not sure if cloning inside a loop is a good idea.

We're keeping the initial client instance for the background task, and clone it once for the blocking calls.